### ISSUE TITLE
Provide Vagrant or Docker file for easy setup of test server.

### ISSUE THREAD

**Issue by PyroLagus** [standardebooks/web#17]
I think it would be great to have a Vagrant/Docker file that automatically sets up a test server, because that would really make things easier for newcomers. Especially with small changes like #16, setting up the server is more work than the actual contribution itself. The Vagrant/Docker setup files would also provide more detailed and complete instructions on how to set up the server (at least on Ubuntu) than the Readme.

**Comment by PyroLagus** [comment]
It would also be nice to have an easy way to clone all the ebook sources collectively. Maybe a repository that has all the book repositories as submodules would be feasible?

**Comment by acabal** [comment]
I'm not familiar with Vagrant so someone would have to set that up and also maintain it. If it's just a single file, then pull requests are welcomed. If it's something more complex then we should probably put that aside for now. The best bet is to install an Ubuntu 18.04 VM and run Apache/PHP-FPM from within it. Apache/FPM config is simple and there are many examples online on how to set it up in just a few lines.

**Comment by PyroLagus** [comment]
I've finished writing a Vagrantfile plus provision script that will set up a server on Ubuntu 18.04 running nginx and php-fpm. I also added a section to the readme, but I'm not sure if its clear enough or too wordy, so I'd appreciate if you could review it. https://github.com/PyroLagus/standardebooks-web/tree/vagrant-ubuntu

**Comment by acabal** [comment]
OK, looks good. Some notes:

- In the readme, there seem to be a few misclosed code blocks in the section starting with "On its first invocation, "

- Instead of naming the provision script `scripts/provision.sh`, let's create a folder under scripts called `vagrant` and then within that folder call the `provision`. So `scripts/vagrant/provision`. There may eventually be other scripts in the `scripts` folder so better to start things organized.

- In `provision.sh`, I see you add a PPA for PHP 7.3 But, we specifically want to use the version of PHP included with Ubuntu 18.04, which is 7.2. Can you remove that PPA and update the rest of the config?

**Comment by PyroLagus** [comment]
Ah, right. I thought 7.2 didn't work, but the problem just ended up being missing dependencies, will fix.
